resolving/npm-resolver/src/pickPackageFromMeta.ts

```diff
@@ -33,7 +34,11 @@ export function pickPackageFromMeta (
       version = spec.fetchSpec
       break
     case 'tag':
-      version = meta['dist-tags'][spec.fetchSpec]
+      if (cmdFullName === 'update' && preferredVersionSelectors && spec.fetchSpec === 'latest') {
+        version = pickLatestVersionToUpdate(meta, preferredVersionSelectors)
+      } else {
+        version = meta['dist-tags'][spec.fetchSpec]
+      }
       break
```

Comment:
Don't change this only for tag. Change it for all types of selectors. Maybe it can be moved outside of this function then.

Reply:
I have changed it so that the version would be reverted as long as `preventDowngrade` is `true`.